### ISSUE TITLE
comparing nested array with compareArraysInOrder: false

### ISSUE THREAD

**Issue by bwgjoseph** [netilon/differify#25]
Hi,

I'm am trying out the library with nested array dataset, but it seem like it doesn't work as I've expected. Is this the correct behavior?

I refer to this [example](https://github.com/netilon/differify#array-comparison-example-without-having-the-order-in-mind-comparearraysinorder-false-)

```js
const differify = new Differify({
	compareArraysInOrder: false,
	mode: { object: 'DIFF', array: 'DIFF' },
});

const diff = differify.compare(['a', 'b'], ['a', 'z', 'b']);

/*
	OUTPUT


	{
		"_": [{
			"original": "a",
			"current": "a",
			"status": "EQUAL",
			"changes": 0
		}, {
			"original": "b",
			"current": "b",
			"status": "EQUAL",
			"changes": 0
		}, {
			"original": null,
			"current": "z",
			"status": "ADDED",
			"changes": 1
		}],
		"status": "MODIFIED",
		"changes": 1
	}

*/
```

This is the dataset I am testing on

```js
const before = {
  members: [
    {
      memberIds: ['a', 'b'],
      originalMemberId: '1',
      type: 'COUNTRYCLUB'
    }
  ]
}

const after = {
  members: [
    {
      memberIds: ['a', 'z', 'b'],
      originalMemberId: '1',
      type: 'COUNTRYCLUB'
    }
  ]
}
```

And this is how I test

```js
const diff = new Differify({
  compareArraysInOrder: false,
  mode: {
    object: DIFF_MODES.DIFF,
    array: DIFF_MODES.DIFF,
  }
})

const dif = diff.compare(before, after);

// output

{
  "_": {
    "members": {
      "_": [
        {
          "original": {
            "memberIds": [
              "a",
              "b"
            ],
            "originalMemberId": "1",
            "type": "COUNTRYCLUB"
          },
          "current": null,
          "status": "DELETED",
          "changes": 1
        },
        {
          "original": null,
          "current": {
            "memberIds": [
              "a",
              "z",
              "b"
            ],
            "originalMemberId": "1",
            "type": "COUNTRYCLUB"
          },
          "status": "ADDED",
          "changes": 1
        }
      ],
      "status": "MODIFIED",
      "changes": 2
    }
  },
  "status": "MODIFIED",
  "changes": 2
}
```

If I were to set `compareArraysInOrder` to `true`

```js
// output

{
  "_": {
    "members": {
      "_": [
        {
          "_": {
            "memberIds": {
              "_": [
                {
                  "original": "a",
                  "current": "a",
                  "status": "EQUAL",
                  "changes": 0
                },
                {
                  "original": "b",
                  "current": "z",
                  "status": "MODIFIED",
                  "changes": 1
                },
                {
                  "original": null,
                  "current": "b",
                  "status": "ADDED",
                  "changes": 1
                }
              ],
              "status": "MODIFIED",
              "changes": 2
            },
            "originalMemberId": {
              "original": "1",
              "current": "1",
              "status": "EQUAL",
              "changes": 0
            },
            "type": {
              "original": "COUNTRYCLUB",
              "current": "COUNTRYCLUB",
              "status": "EQUAL",
              "changes": 0
            }
          },
          "status": "MODIFIED",
          "changes": 2
        }
      ],
      "status": "MODIFIED",
      "changes": 2
    }
  },
  "status": "MODIFIED",
  "changes": 2
} 
```

Here's the [codesandbox](https://codesandbox.io/s/interesting-currying-xzd3n?file=/src/index.js)

What I wanted was to be able to compare `memberIds` correctly and output that `a, b` are `EQUAL` and `z` is `ADDED`. How can I achieve that?

Thank you!

**Comment by fabian-orue** [comment]
Hi @bwgjoseph , thank you for let me know. I will check it.

**Comment by bwgjoseph** [comment]
Thanks, @fabian-orue!

**Comment by fabian-orue** [comment]
Hi @bwgjoseph , this issue should be fixed in the recently released version (**3.3.3**). Thank you for yout help! Regards.
